sklearn/calibration.py

```diff
@@ -431,19 +428,26 @@ def _fit_classifier_calibrator_pair(estimator, X, y, train, test, supports_sw,
     -------
     calibrated_classifier : _CalibratedClassifier instance
     """
-    if sample_weight is not None and supports_sw:
-        estimator.fit(X[train], y[train],
-                      sample_weight=sample_weight[train])
+    X_train, y_train = _safe_indexing(X, train), _safe_indexing(y, train)
+    X_test, y_test = _safe_indexing(X, test), _safe_indexing(y, test)
+    if sample_weight is not None:
```

Comment:
Since we are not strict about sample weights + estimators that do not support sample weights:

https://github.com/scikit-learn/scikit-learn/blob/80f923e00d6949b2385612b024981ac78a79e45a/sklearn/calibration.py#L287-L291

We do not need to do this indexing when `supports_sw` is False.

```suggestion
    if supports_sw and sample_weight is not None:
```